### PR TITLE
[Go] Support multiple return values matching (#4896)

### PR DESCRIPTION
**Pull request by bmahe**
PR checklist:

- [x] Documentation is up-to-date
- [X] Changelog is up-to-date
- [X] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by bmahe** [comment]
I am not aware of any doc to update for that change, but please let me know if there is any

**Comment by aryx** [comment]
This closes #4896

**Review by aryx — APPROVED** [review]
Nice.

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
   tree-sitter-typescript (Oct 2021)
 - TS: support for override keyword (#4220, #4798)
 - TS: better ASI (#4459) and accept (null)(foo) (#4468)
+- Go: Support multiple return values matching (#4896)
```
Comment (aryx) [inline CHANGELOG.md]: You can maybe say "support for ellipsis in multiple return values (e.g., func foo() (..., error, ...) {}) (#4896)"

File: semgrep-core/tests/go/dots_type_tuple.go [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+func test3() (int, int, error) {
+	return 0, 1, nil
+}
+
+func test4() (int, error, string) {
+	return 0, nil, "hello"
+}
+
+/**
+ * error is parsed as a single type and thus would not count as a tuple
```
Comment (aryx) [inline semgrep-core/tests/go/dots_type_tuple.go:22]: nice comment

**Review by aryx — APPROVED** [review]

**Pull request merged**